Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -260,109 +278,139 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
-        /// <para>              Try to use feed from local package, if feed from local package also null, fallback to default feed</para>
-        /// <para>              Check if version from query is same as local package</para>
-        /// </summary>
-        private async Task<bool> IsSiteExtensionInstalled(string id, string version, string feedUrl)
-        {
-            JsonSettings siteExtensionSettings = GetSettingManager(id);
-            string localPackageVersion = siteExtensionSettings.GetValue(_versionSetting);
-            string localPackageFeedUrl = siteExtensionSettings.GetValue(_feedUrlSetting);
-            bool isInstalled = false;
 
-            // Try to use given feed
-            // If given feed is null, try with feed that from local package
-            // And GetRemoteRepository will fallback to use default feed if pass in feed param is null
-            SourceRepository remoteRepo = GetRemoteRepository(feedUrl ?? localPackageFeedUrl);
-
-            // case 1 and 2
-            if (!string.IsNullOrWhiteSpace(version)
-                && version.Equals(localPackageVersion, StringComparison.OrdinalIgnoreCase)
-                && remoteRepo.PackageSource.Source.Equals(localPackageFeedUrl, StringComparison.OrdinalIgnoreCase))
+            if (info == null)
             {
-                isInstalled = true;
+                info = new SiteExtensionInfo();
+                info.ProvisioningState = Constants.SiteExtensionProvisioningStateFailed;
+                info.Comment = string.Format(Constants.SiteExtensionProvisioningStateNotFoundMessageFormat, id);
+                status = HttpStatusCode.NotFound;
             }
-            else if (string.IsNullOrWhiteSpace(version) && string.IsNullOrWhiteSpace(feedUrl))
+            else if (!string.Equals(Constants.SiteExtensionProvisioningStateFailed, info.ProvisioningState, StringComparison.OrdinalIgnoreCase))
             {
-                // case 3
-                UIPackageMetadata remotePackage = await remoteRepo.GetLatestPackageById(id);
-                if (remotePackage != null)
-                {
-                    isInstalled = remotePackage.Identity.Version.ToNormalizedString().Equals(localPackageVersion, StringComparison.OrdinalIgnoreCase);
-                }
+                info.ProvisioningState = Constants.SiteExtensionProvisioningStateSucceeded;
+                info.Comment = null;
             }
 
-            return isInstalled;
+            using (tracer.Step("Update arm settings for {0} installation", id))
+            {
+                SiteExtensionArmSettings armSettings = await SiteExtensionArmSettings.GetSettings(_environment.SiteExtensionSettingsPath, id);
+                armSettings.ReadSiteExtensionInfo(info);
+                armSettings.Status = status;
+                armSettings.Operation = alreadyInstalled ? null : Constants.SiteExtensionOperationInstall;
+                await armSettings.SaveArmSettings();
```

Comment:
What if this is unable to Write to file - will the status stay pending forever?   Is there any correction path?